src/runtime/metal.cpp

```diff
@@ -239,7 +247,8 @@ WEAK int halide_metal_acquire_context(void *user_context, mtl_device *&device_re
         }
     }
 
-    // If the context has not been initialized, initialize it now.
+    // If the device has already been initialized,
+    // ensure the queue has as well.
     halide_assert(user_context, queue != 0);
```

Comment:
I think a better fix is to delete the code above and change this assert to:
    halide_assert(user_context, (device == 0) || (queue != 0));